### PR TITLE
Add a few more privacy consideration sections

### PR DESCRIPTION
**Pull request by csharrison**
***
<a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/pull/816.html" title="Last updated on May 18, 2023, 8:46 PM UTC (4b12a26)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/816/84aa0c2...4b12a26.html" title="Last updated on May 18, 2023, 8:46 PM UTC (4b12a26)">Diff</a>

### CODE REVIEW

**Comment by csharrison** [comment]
PTAQL. This is mostly cribbed from the explainers but I've tried to make things a bit more precise with the channel capacity stuff.

File: index.bs [hunk 1]
```diff
@@ -3070,10 +3070,99 @@ Issue: Fill in this section.
...
+## Cross-site information disclosure ## {#cross-site-information-disclosure}
+
+The API is concerned with protecting arbitrary cross-site information from being passed
+from one site to another. For a given [=attribution source=], any outcome associated with it
+is considered cross-site information. This includes:
+* Whether the [=attribution source=] generates any [=attribution reports=] or not
+* The contents of the associated [=attribution reports=], if present
+
+The information embedded in the API output is arbitrary but can include things like browsing
+history and other cross site activity. The API aims to provide some protection for this information:
```
Comment (apasel422) [inline index.bs:3087]:
```suggestion
history and other cross-site activity. The API aims to provide some protection for this information:
```

File: index.bs [hunk 2]
```diff
@@ -3070,10 +3070,99 @@ Issue: Fill in this section.
...
+
+### Event-level reports ### {#cross-site-information-disclosure-event-level-reports}
+
+Any given [=attribution source=] has a [=obtain a set of possible trigger states|set of possible trigger states=].
+The choice of trigger state may encode cross site information. To protect the cross-site information disclosure,
```
Comment (apasel422) [inline index.bs:3092]:
```suggestion
The choice of trigger state may encode cross-site information. To protect the cross-site information disclosure,
```

File: index.bs [hunk 3]
```diff
@@ -3070,10 +3070,99 @@ Issue: Fill in this section.
...
+each [=attribution source=] is subject to a [=obtain a randomized response|randomized response mechanism=],
+which will choose a state at random with [=obtain a randomized source response pick rate|pick rate=]
+dependent on the user agent's [=randomized response epsilon=].
+
+This introduces some level of plausible deniability into the resulting [=event-level reports=] (or lack thereof),
+as there is always a chance that the output was generated from a random process. We can reason about the
+protection this gives an individual [=attribution source=] from the lens of differential privacy.
+
+Additionally, [=event-level reports=] limit the *amount* of relative cross-site information associated with a
```
Comment (apasel422) [inline index.bs:3101]:
```suggestion
Additionally, [=event-level reports=] limit the *amount* of relative cross-site information associated with
```

File: index.bs [hunk 4]
```diff
@@ -3070,10 +3070,99 @@ Issue: Fill in this section.
...
+a particular [=attribution source=]. We model this using the notion of channel capacity.  For every [=attribution source=],
+it is possible to model its output as a noisy channel. The number of input/output symbols is governed by it's associated
```
Comment (apasel422) [inline index.bs:3103]:
```suggestion
it is possible to model its output as a noisy channel. The number of input/output symbols is governed by its associated
```

File: index.bs [hunk 5]
```diff
@@ -3070,10 +3070,99 @@ Issue: Fill in this section.
...
+[=obtain a set of possible trigger states|set of possible trigger states=].
+
+The channel capacity of a q-ary symmetric channel is equal to `log2(q) - h(p) - p log2(q - 1)` where
+`h` is the binary entropy function and `p` is the [=obtain a randomized source response pick rate|pick rate=].
+
+Note that [=source type/navigation=] [=attribution sources=] and [=source type/event=] [=attribution sources=] may
+have different channel capacities, given that [=source type/event=] [=attribution sources=] can be registered without
+[=user activation=] or top-level navigation.
+
+Issue: Link to a definition of channel capacity, q-ary symmetrics channel, binary entropy function, randomized response, and
```
Comment (apasel422) [inline index.bs:3115]:
```suggestion
Issue: Link to a definition of channel capacity, q-ary symmetric channel, binary entropy function, randomized response, and
```

File: index.bs [hunk 6]
```diff
@@ -3070,10 +3070,99 @@ Issue: Fill in this section.
...
+
+A primary privacy goal of the API is to make linking identity between two different top-level sites
+difficult. This happens when either a request or a JavaScript environment has two user IDs from two
+different sites simultaneously. Both [=event-level reports=] and [=aggregatable reports=] were
+designed to make this kind of recognition difficult:
+
+### Event-level reports ### {#cross-site-recognition-event-level-reports}
+[=Event-level reports=] come bearing a fine-grained [=event-level report/event id=] that can
+uniquely identify the source event, which may be joinable with a user's identity. As such, for
+[=event-level reports=] to protect the cross site recognition risk, they contain only a small
```
Comment (apasel422) [inline index.bs:3144]:
```suggestion
[=event-level reports=] to protect the cross-site recognition risk, they contain only a small
```

File: index.bs [hunk 7]
```diff
@@ -3070,10 +3070,99 @@ Issue: Fill in this section.
...
+amount (measured via channel capacity) of relative cross-site information from any of the
+[=attribution source/attribution destinations=]. By limiting the amount of relative cross-site
+information embedded in [=event-level reports=], we make it difficult for an identifier to be passed
+through this channel to enable cross-site recognition.
+
+### Aggregatable reports ### {#cross-site-recognition-aggregatable-reports}
+[=Aggregatable reports=] only contain fine-grained cross-site information in encrypted form.
+In cleartext, they contain only coarse grained information from the [=attribution source/source site=]
```
Comment (apasel422) [inline index.bs:3152]:
```suggestion
In cleartext, they contain only coarse-grained information from the [=attribution source/source site=]
```

File: index.bs [hunk 8]
```diff
@@ -3070,10 +3070,99 @@ Issue: Fill in this section.
...
+and [=aggregatable report/effective attribution destination=]. This makes it difficult for an
+[=aggregatable report=] to be associated with a user from either site.
+
+The cross site recognition risk of the data encrypted in "`aggregation_service_payloads`" is
```
Comment (apasel422) [inline index.bs:3156]:
```suggestion
The cross-site recognition risk of the data encrypted in "`aggregation_service_payloads`" is
```

File: index.bs [hunk 9]
```diff
@@ -3070,10 +3070,99 @@ Issue: Fill in this section.
...
+[=Event-level reports=] come bearing a fine-grained [=event-level report/event id=] that can
+uniquely identify the source event, which may be joinable with a user's identity. As such, for
+[=event-level reports=] to protect the cross site recognition risk, they contain only a small
+amount (measured via channel capacity) of relative cross-site information from any of the
+[=attribution source/attribution destinations=]. By limiting the amount of relative cross-site
+information embedded in [=event-level reports=], we make it difficult for an identifier to be passed
+through this channel to enable cross-site recognition.
+
+### Aggregatable reports ### {#cross-site-recognition-aggregatable-reports}
+[=Aggregatable reports=] only contain fine-grained cross-site information in encrypted form.
```
Comment (apasel422) [inline index.bs:3151]:
```suggestion

[=Aggregatable reports=] only contain fine-grained cross-site information in encrypted form.
```

File: index.bs [hunk 10]
```diff
@@ -3070,10 +3070,99 @@ Issue: Fill in this section.
...
+
+## Protecting against cross-site recognition ## {#protecting-against-cross-site-recognition}
+
+A primary privacy goal of the API is to make linking identity between two different top-level sites
+difficult. This happens when either a request or a JavaScript environment has two user IDs from two
+different sites simultaneously. Both [=event-level reports=] and [=aggregatable reports=] were
+designed to make this kind of recognition difficult:
+
+### Event-level reports ### {#cross-site-recognition-event-level-reports}
+[=Event-level reports=] come bearing a fine-grained [=event-level report/event id=] that can
```
Comment (apasel422) [inline index.bs:3142]:
```suggestion

[=Event-level reports=] come bearing a fine-grained [=event-level report/event id=] that can
```

**Review by apasel422 — APPROVED** [review]

File: index.bs [hunk 11]
```diff
@@ -3070,10 +3070,101 @@ Issue: Fill in this section.
...
+
+### Event-level reports ### {#cross-site-recognition-event-level-reports}
+
+[=Event-level reports=] come bearing a fine-grained [=event-level report/event id=] that can
+uniquely identify the source event, which may be joinable with a user's identity. As such, for
+[=event-level reports=] to protect the cross-site recognition risk, they contain only a small
+amount (measured via channel capacity) of relative cross-site information from any of the
+[=attribution source/attribution destinations=]. By limiting the amount of relative cross-site
+information embedded in [=event-level reports=], we make it difficult for an identifier to be passed
+through this channel to enable cross-site recognition.
```
Comment (johnivdel) [inline index.bs:3149]: one idea, feel free to ignore:

```suggestion
through this channel to enable cross-site recognition alone.
```

File: index.bs [hunk 12]
```diff
@@ -3070,10 +3070,101 @@ Issue: Fill in this section.
 
 # Privacy considerations # {#privacy-considerations}
 
+## Clearing site data ## {#clearing-site-data}
+
 The [=attribution caches=] contain data about a user's web activity. As such,
 the user agent MAY expose controls that allow the user to delete data from them.
 
-Issue: Discuss general principles behind design decisions for the API.
+## Cross-site information disclosure ## {#cross-site-information-disclosure}
```
Comment (johnivdel) [inline index.bs:3078]: Do sections like this typically get marked non-normative? E.g. I was looking at WebBluetooth as a spot check (https://webbluetoothcg.github.io/web-bluetooth/#privacy) and quite a bit if marked non-normative.

**Review by johnivdel — APPROVED** [review]
LGTM with a question

File: index.bs (hunk 11 above)
Comment (csharrison) [reply index.bs]: Done

File: index.bs (hunk 12 above)
Comment (csharrison) [reply index.bs:3078]: It seems reasonable to do this.

**Review by johnivdel — APPROVED** [review]
SLGTM

**Pull request merged**